haystack/reader/base.py

```diff
@@ -66,7 +66,6 @@ def run(self, query: str, documents: List[Document], top_k_reader: Optional[int]
                 if doc.id == ans["document_id"]:
                     ans["meta"] = deepcopy(doc.meta)
 
-        results.update(**kwargs)
         return results, "output_1"
```

Comment:
I think we wanted to adjust the return signature here further to include `debug` and `output_1` in one typed dict called `ComponentResponse` ?

Reply:
With the updated implementation, a node can return a `_debug` value of any type(str, dict, etc).

This `_debug` gets appended to the global `_debug` for the request with corresponding node names.

For instance,

```python
class MyComponent:
    def run(....):
        output = {..., "_debug": "my debug information ..."}  # the _debug key can also be a dict
        return output, "output_1"
```
will get transformed in final response like,

```JSON
{"answers": ..., "_debug": {"MyComponent": "my debug information ..."}}

```